### PR TITLE
Updated Gradle and Android build tools versions

### LINKED ISSUE #408: Update Gradle version and Android Gradle build tools
Android is currently using `com.android.tools.build:gradle:4.1.3` for building `ably_flutter`. The current stable version of this package is `7.2.0`, and it corresponds to the same version of the Gradle build tool itself. Since newer versions of Gradle come with a lot of benefits (mainly shorter build times) we should consider updating the build tools and Gradle to a more recent version

┆Issue is synchronized with this [Jira Uncategorised](https://ably.atlassian.net/browse/SDK-1900) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by ikurek**
This closes #408. I've updated Gradle version to the latest one available `7.4.2` and Android build tools to `7.2.0`. I've also removed deprecated references to `jCenter()` and set both compile and target SDK versions to the latest Android SDK - `32`.

### CODE REVIEW

**Comment by ikurek** [comment]
I've switched this to draft, until https://github.com/ably/ably-flutter/issues/417 is resolved

File: example/android/gradle.properties [hunk 1]
```diff
@@ -1,4 +1,3 @@
 org.gradle.jvmargs=-Xmx1536M
-android.enableR8=true
```
Comment (KacperKluka) [inline example/android/gradle.properties:2]: Why are we disabling the R8? Is it causing some issues? Or is it enabled by default, so there's no need to do it explicitly? :thinking:

File: test_integration/android/gradle.properties [hunk 2]
```diff
@@ -1,4 +1,3 @@
 org.gradle.jvmargs=-Xmx1536M
-android.enableR8=true
```
Comment (KacperKluka) [inline test_integration/android/gradle.properties:2]: The same question as above :wink:

File: example/android/gradle.properties (hunk 1 above)
Comment (ikurek) [reply example/android/gradle.properties:2]: The `android.enableR8` flag is deprecated with the latest versions of Gradle

File: test_integration/android/gradle.properties (hunk 2 above)
Comment (ikurek) [reply test_integration/android/gradle.properties:2]: The same answer as above 😆

**Review by KacperKluka — APPROVED** [review]
LGTM :+1:

**Review by QuintinWillison — APPROVED** [review]

**Review by ikbalkaya — APPROVED** [review]
Skimmed through it. But lgtm

**Pull request merged**